doc/developer/index.adoc

```diff
@@ -26,29 +26,136 @@ These conditions are to ensure that existing user code will continue to work whe
 
 For patch releases we never allow changes in the public API, unless the change is specifically to fix a bug that aligns the actual behavior of the code with the publicly documented behavior.
 
+Only release branches (more on those later) include the full major.minor.patch
+version numbers included with releases. The major and minor version of the _master_ and
+_develop_ branches do not include a patch number and always have a SNAPSHOT tag.
+The _master_ and _develop_ versions always matches the pattern *MAJOR*.*MINOR*-SNAPSHOT.
+
+The _master_ version always has the same major and minor number as the latest
+release while the _develop_ version uses the next expected major/minor
+release number, such as 2.1-SNAPSHOT and 2.2-SNAPSHOT respectively
+(after a 2.1.x release, but before any 2.2.x releases are made public).
+
 == Workflow
 
-RDF4J uses a git branching model where feature development takes place on branches from the master branch. This is where all development for the next (minor or major) release happens.
+RDF4J uses a git branching model where feature development takes place on branches from the _develop_ branch. This is where all development for the next (minor or major) release happens.
+These branches should have the prefix `features/` to indicate they are either
```

Comment:
The first use of "features/" here is still ambiguous when reading through the document from top to bottom, as the specifics are described below this point, and "issues/#" is first introduced below here. It isn't clear why you need to introduce the "features/" convention, (which has a high likelihood of only being used a few times a year) here, given that the other 90% of PRs will be compact and use "issues/#" straight from develop or master. However, feel free to merge this and incrementally fix up bits like that.